File: cases/taint/kotlin/local-overwrite-positive/LocalOverwritePositive.kt
package dataflowbench

object LocalOverwritePositive {
    fun dfb_source(): Int { // DFB-SOURCE: local-overwrite-input
        return 1
    }

    fun dfb_sink(value: Int) {} // DFB-SINK: local-overwrite-sink

    fun run() {
        var value = dfb_source()
        value = value // DFB-WITNESS: local-overwrite-preserved
        dfb_sink(value)
    }
}

File: cases/taint/php/local-overwrite-positive/local_overwrite_positive.php
<?php
function dfb_source(): string { // DFB-SOURCE: local-overwrite-input
    return "tainted";
}

function dfb_sink(string $value): void {} // DFB-SINK: local-overwrite-sink

function run(): void {
    $value = dfb_source();
    $value = $value; // DFB-WITNESS: local-overwrite-preserved
    dfb_sink($value);
}
